### DISCUSSION TITLE
How are blocking/expensive operations handled without blocking other requests?

### DISCUSSION THREAD

**Discussion by conways-glider** [tokio-rs/axum#1259] [Q&A] [answered]
I'm looking to understand if this library has an equivalent to actix-web's `web::block`. 

If not, how does axum prevent long running requests from blocking new requests?
Does tokio just handle it, or do I need to spawn a new tokio thread?

**Comment by davidpdrsn** [comment, marked as answer]
Use https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html

**Reply by conways-glider** [reply]
Thank you for answering! I wonder if adding an example of axum using [`spawn_blocking`](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html) may be a good idea. That way, there's easy documentation coming into this repo with little tokio knowledge.

**Reply by davidpdrsn** [reply]
Where would you add that? It's quite common when using tokio/async rust in general.

**Reply by conways-glider** [reply]
I was thinking giving it it's own example in the axum examples or adding it to this one: https://github.com/tokio-rs/axum/tree/main/examples/sqlx-postgres

I figured it was probably common in tokio, but I didn't know it was common in async rust in general - is that implemented in std or the futures crate as well? Pardon my ignorance - Just haven't worked with async rust or tokio much.

**Reply by davidpdrsn** [reply]
> is that implemented in std or the futures crate as well?

Nope. It requires runtime support so must be built into tokio. futures or std don't have async runtimes.

I don't think this is worth documenting in axum. I'd recommend reading the [tokio tutorial](https://tokio.rs/tokio/tutorial) when getting started.

**Reply by conways-glider** [reply]
Ooh! That's a well written tutorial! Thank you for the resource!
